src/commands/docs/mdn.ts

```diff
@@ -1,145 +1,131 @@
-    const embed = this.client.util
-      .embed()
-      .setColor(this.MDN_BLUE_COLOR)
-      .setAuthor("MDN Documentation", this.MDN_ICON_URL)
-      .setTitle(`Search for: ${query}`);
+    const res = await fetch("https://developer.mozilla.org/sitemaps/en-us/sitemap.xml.gz");
+    if (!res.ok) return sources; // Fallback to old sources if the new ones are not available for any reason
 
-    if (!search.length) {
-      embed.setColor(0xFF0000).setDescription("No results found...");
-      return message.util.send(embed);
-    }
+    const sitemap: Sitemap<number> = new XMLParser()
+        .parse(gunzipSync(await res.buffer()).toString())
+        .urlset.url.map((entry: SitemapEntry<string>) => ({
+            loc: entry.loc.slice(baseURL.length),
+            lastmod: new Date(entry.lastmod).valueOf(),
+        }));
 
-    if (search.length === 1) {
-      const res = await fetch(`${baseURL + search[0]}/index.json`);
-      const doc: MdnDoc = (await res.json()).doc;
-      const docEmbed = this.client.util
-        .embed()
-        .setColor(0xFFFFFF)
-        .setTitle(doc.pageTitle)
-        .setURL(`https://developer.mozilla.org/${doc.mdn_url}`)
-        .setThumbnail(this.MDN_ICON_URL)
-        .setDescription(doc.summary);
-      return message.util.send(docEmbed);
-    }
+    const index = new flexsearch.Index();
+    sitemap.forEach((entry, idx) => index.add(idx, entry.loc));
 
-    let results = search.map((path) => `**• [${path.replace(/_|-/g, " ")}](${baseURL}${path})**`);
-    embed.setDescription(results.join("\n"));
-    return message.util.send(embed);
-  }
+    sources = { index, sitemap, lastUpdated: Date.now() };
+    return sources;
 }
 
 interface MdnDoc {
-  isMarkdown: boolean,
-  isTranslated: boolean,
-  isActive: boolean,
-  flaws: {},
-  title: string,
-  mdn_url: string,
-  locale: string,
-  native: string,
-  sidebarHTML: string,
-  body: ({
-    type: "prose" | "specifications" | "browser_compatibility",
-    value: {
-      id: string | null,
-      title: string | null,
-      isH3: boolean,
-      // type:prose
-      content?: string,
-      // type:specifications
-      specifications?: ({ bcdSpecificationURL: string, title: string, shortTitle: string })[],
-      // type:browser_compatibility
-      dataURL?: string,
-      // type:specifications | type:browser_compatibility
-      query?: string,
-    }
-  })[],
-  toc: ({ text: string, id: string })[],
-  summary: string,
-  popularity: number,
-  modified: string, // ISO Date String
-  other_translations: ({ title: string, locale: string, native: string })[],
-  source: {
-    folder: string,
-    github_url: string,
-    last_commit_url: string,
-    filename: string
-  },
-  parents: ({ uri: string, title: string })[],
-  pageTitle: string,
-  noIndexing: boolean
+    isMarkdown: boolean;
+    isTranslated: boolean;
+    isActive: boolean;
+    //TODO Fix this
```

Comment:
I am unaware of how this should be typed
however since now eslint rules work, this should be resolved, you should use the proper typing for this

Reply:
Due to lack of documentation on MDN's JSON format, the type of that object is unknown other than it being an object, so I just changed it to `Record<string, unknown>` for now.